### PR TITLE
adding a validation to remove all break line and tabs from a string f…

### PR DESCRIPTION
**Pull request by pmoraes**
Hello guys, 

This pull request is to solve this issue https://github.com/cakephp/cakephp/issues/8179.

By any problems, please let me know.

Thanks, Peter Moraes.

### CODE REVIEW

File: lib/Cake/Model/Datasource/DboSource.php [hunk 1]
```diff
@@ -2550,14 +2550,17 @@ public function fields(Model $Model, $alias = null, $fields = array(), $quote =
...
+		$count = count($fields);
 		if ($count >= 1 && !in_array($fields[0], array('*', 'COUNT(*)'))) {
 			for ($i = 0; $i < $count; $i++) {
 				if (is_string($fields[$i]) && in_array($fields[$i], $virtual)) {
 					unset($fields[$i]);
 					continue;
 				}
+
+				$fields[$i] = preg_replace('/\t/', '', $fields[$i]);
+				$fields[$i] = str_replace(array("\r", "\n"), '', $fields[$i]);
```
Comment (markstory) [inline lib/Cake/Model/Datasource/DboSource.php]: What about literal newlines that are embedded in a CONCAT() expression?

Comment (pmoraes) [inline lib/Cake/Model/Datasource/DboSource.php]: I have made a new commit removing the line 2562. so it just will remove the break lines then the query will stay in the same line and this regex https://github.com/CakeDC/cakephp/blob/hot-fix/distinct-with-break-line/lib/Cake/Model/Datasource/DboSource.php#L2567 won't match with with the query with break line.

**Comment by dereuromark** [comment]
My concern remains:

> Couldn't that potentially alter values that contain newlines (on purpose)?
I think there is nothing we can really do here without creating other potentially BC breaking side-effects.

**Comment by ADmad** [comment]
This is a risky change. :-1:

**Comment by jhli** [comment]
If there is no good way out of this, documentation in book and api is king.

**Comment by ravage84** [comment]
:-1: yep, too risky. I agree with @jhli

**Pull request closed without merging**